src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -754,6 +756,32 @@ protected DynamoViewModel(StartConfiguration startConfiguration)
             FileTrustViewModel = new FileTrustWarningViewModel();
         }
 
+        private void CurrentDispatcher_UnhandledException(object sender, DispatcherUnhandledExceptionEventArgs e)
+        {
+            if (e.Handled)
+            {
+                return;
+            }
+
+            e.Handled = true;
+            CrashGracefully(e.Exception);
```

Comment:
so I'm a bit confused.

Whats the value of this handler? Maybe thats what @pinzart90 was mentioning today.
For example, it looks like we always call `CrashGracefully` - do we intend to not crash and just log to the console in some cases? For example if the call stack indicates an exception from a view extension?

Reply:
The value of this handler is there for hosts other than Revit that do not handle these exceptions and that do not use DynamoSandbox ( thus they don't have the try-catch around RunApplication either).

Reply:
And yes, using this handler we now have the opportunity to just long and continue for some types of exceptions (like from a view extension) (also a benefit or consequence would be to not catch non-Dynamo related exceptions )

Reply:
@pinzart90 do you mean:
1. we should not shutdown Dynamo here? At the moment it is shutting down Dynamo.
2. this handler will NOT catch non-Dynamo exceptions? I thought we discussed earlier that this will continue to catch non-Dynamo exceptions.

Reply:
For both points I was only saying we have an opportunity for the future. I did not mean to imply that we should do it or that it is already working like that.